lib/hsa/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
+include(FindPkgConfig)
+
 ####################
 # HCC runtime (HSA implementation)
 ####################
 if (HAS_ROCM EQUAL 1)
 add_mcwamp_library_hsa(mcwamp_hsa mcwamp_hsa.cpp unpinned_copy_engine.cpp)
-target_link_libraries(mcwamp_hsa  unwind)
+pkg_check_modules(LIBUNWIND REQUIRED libunwind)
+# TODO: we should probably add required CFLAGS and LDFLAGS from pkgconfig
+target_link_libraries(mcwamp_hsa ${LIBUNWIND_LIBRARIES})
```

Comment:
> Why is this necessary for hcc configure path if the imported properties are only used in downstream libraries?

The imported library is created in the hcc-config.cmake file as well. This allows for the dependencies to be relocatable. Otherwise hardcoded paths are used in the `mcwamp_hsa` target. Downstream libraries would use these hardcoded paths, so its better to create an imported target, so that hardcoded paths are not used.

> also shouldn't the add_library use PUBLIC instead of INTERFACE since both hcc and downstream libraries need to link to it?

There is no `PUBLIC` keyword that can be used when using `add_library`. This is an interface-only library just to store the flags from pkgconfig needed to use libunwind.